content/en/docs/guides/keepalive.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Keepalive
+description: >-
+  How to use HTTP/2 PING-based keepalives in gRPC
+---
+
+### Overview
+
+HTTP/2 PING-based keepalives are a way to keep a HTTP/2 connection alive even when there is no data being transferred. This is done by periodically sending a PING frame to the other end of the connection. HTTP/2 keepalives can improve performance and reliability of HTTP/2 connections, but it is important to configure the keepalive interval carefully.
+
+{{% alert title="Note" color="info" %}}
+  There is a related but separate concern called [Health Checking]. Health checking allows server to signal whether a *service* is healthy while keepalive is only about the *connection*.
+{{% /alert %}}
+
+### Background
+
+[TCP keepalive] is a well-known method of maintaining connections and detecting broken connections. When enabled, either side of the connection can send redundant packets, once ACKed by the other side, the connection will be consiered as good. If no ACK after repeated attempts, the connection is deemed broken. Configuration of its variables is provided by the OS on a per-socket level, but is commonly not exposed to higher-level APIs.
+
+TCP keepalive has three parameters to tune:
+
+1. Time (time since last receipt before sending a keepalive).
+2. Interval (interval between keepalives when not receiving reply).
+3. Retry (number of times to retry sending keepalives).
+
+gRPC uses HTTP/2 which provides a mandatory [PING] frame which can be used to estimate round-trip time, bandwidth-delay product, or test the connection. The interval and retry in TPC keepalive don't quite apply to PING because the transport is reliable, so they're replaced with timeout (equivalent to interval * retry) in gRPC PING-based keepalive implementation.
+
+{{% alert title="Note" color="info" %}}
+  It's not required for service owners to support kepalive. **Client authors must coordinate with service owners** for whether a particular client-side setting is acceptable. Service owners decide what they are willing to support, including whether they are willing to receive keepalives at all.
```

Comment:
When a server is unwilling to receive keepalives, what happens when it receives one? It might be useful to mention that

Reply:
If the service does not support keepalive, the first few keepalive pings will be ignored, and the server will eventually send a `GOAWAY` message with debug data equal to the ASCII code for `too_many_pings`.

Will add this to note.